app/src/main/java/de/danoeh/antennapod/dialog/FilterDialog.java

```diff
@@ -21,42 +29,143 @@ public FilterDialog(Context context, FeedItemFilter feedItemFilter) {
     }
 
-    public void openDialog() {
-        final String[] items = context.getResources().getStringArray(R.array.episode_filter_options);
-        final String[] values = context.getResources().getStringArray(R.array.episode_filter_values);
-        final boolean[] checkedItems = new boolean[items.length];
+    public void openCustomDialog() {
 
         final Set<String> filterValues = new HashSet<>(Arrays.asList(filter.getValues()));
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle(R.string.filter);
 
-        // make sure we have no empty strings in the filter list
-        for (String filterValue : filterValues) {
-            if (TextUtils.isEmpty(filterValue)) {
-                filterValues.remove(filterValue);
-            }
-        }
+        LayoutInflater inflater = LayoutInflater.from(this.context);
+        View customLayout = inflater.inflate(R.layout.custom_filter_dialog_layout, null);
+        TableLayout tableLayout = customLayout.findViewById(R.id.custom_dialog_tableLayout1);
+
+        builder.setView(customLayout);
+
+        for (FeedItemEnum item : FeedItemEnum.values()) {
+
+            RelativeLayout row = (RelativeLayout) inflater.inflate(R.layout.custom_dialog_filter_row, null);
+            TextView textView1 = row.findViewById(R.id.custom_dialog_textView1);
+            TextView textView2 = row.findViewById(R.id.custom_dialog_textView2);
+            TextView textView3 = row.findViewById(R.id.custom_dialog_textView3);
+            textView1.setText(item.values[1].displayName);
+            textView1.setTextColor(Color.DKGRAY);
+            textView2.setText(item.values[0].displayName);
+            textView2.setTextColor(Color.DKGRAY);
 
-        for (int i = 0; i < values.length; i++) {
-            String value = values[i];
-            if (filterValues.contains(value)) {
-                checkedItems[i] = true;
+            Iterator<String> filterIterator = filterValues.iterator();
+            while (filterIterator.hasNext()) {
+                String nextItem = filterIterator.next();
+                if (item.values[1].filterId.equals(nextItem)) {
+                    item.values[1].setSelected(true);
+                    item.values[0].setSelected(false);
+                    textView1.setSelected(true);
+                    textView2.setSelected(false);
+                    textView1.setTextColor(Color.WHITE);
+                    textView2.setTextColor(Color.DKGRAY);
+                    textView3.setBackgroundResource(R.drawable.x_custom_on);
+                }
+                if (item.values[0].filterId.equals(nextItem)) {
+                    item.values[0].setSelected(true);
+                    item.values[1].setSelected(false);
+                    textView2.setSelected(true);
+                    textView1.setSelected(false);
+                    textView2.setTextColor(Color.WHITE);
+                    textView1.setTextColor(Color.DKGRAY);
```

Comment:
The text color can also be specified with a color selector (similar to the backgrounds). That way, you do not need to set it in code. Additionally, this will make it easier to support both light and dark theme. Details: https://stackoverflow.com/a/3565624/4193263